### ISSUE TITLE
Style/ConditionalAssignment incorrectly runs depending on DisabledByDefault

### ISSUE THREAD

**Issue by jletizia** [bbatsov/rubocop#5284]
Running the following command with the following two configurations provides different, unexpected output.

`rubocop --only Style -c <path_to_config> -SD`

--------

## Expected behavior

Not sure which of these behaviors is correct, but I wouldn't expect this configuration option to make a difference here.

## Actual behavior
Using `rubocop --only Style -c <path_to_config> -SD` with the following two configurations, we see differences in output:
 
No violations
```
AllCops:
  TargetRubyVersion: 2.2
  DisabledByDefault: false
  Exclude:
      <omitted>
Style:
  Enabled: true
Style/GlobalVars:
  Enabled: false
Style/ClassVars:
  Enabled: false
Style/Documentation:
  Enabled: false
Style/SingleLineBlockParams:
  Enabled: false
Style/RedundantReturn:
  Enabled: false
Style/RedundantBegin:
  Enabled: false
Style/FormatStringToken:
  Enabled: false
```

Produces `Style/ConditionalAssignment` violations:
```
AllCops:
  TargetRubyVersion: 2.2
  DisabledByDefault: true
  Exclude:
      <omitted>
Style:
  Enabled: true
Style/GlobalVars:
  Enabled: false
Style/ClassVars:
  Enabled: false
Style/Documentation:
  Enabled: false
Style/SingleLineBlockParams:
  Enabled: false
Style/RedundantReturn:
  Enabled: false
Style/RedundantBegin:
  Enabled: false
Style/FormatStringToken:
  Enabled: false
```
## Steps to reproduce the problem
Set `DisabledByDefault` to true / false in the configuration

## RuboCop version

Include the output of `rubocop -V`. Here's an example:

```
$ rubocop -V
0.51.0 (using Parser 2.4.0.0, running on ruby 2.2.4 x86_64-linux-gnu)
```

**Comment by pocke** [comment]
Minimum Reproduce
----


`test.rb`

```ruby
if condcondcondcondcondcondcondcondcondcondcondcondcondcondcondcondcondcondcond
  x = 1
else
  x = 2
end

p x
```

`config.yml`



```yaml
Metrics/LineLength:
  Enabled: false
```


```bash
$ rubocop --only ConditionalAssignment
Inspecting 1 file
.

1 file inspected, no offenses detected



$ rubocop --only ConditionalAssignment --config config.yml
Inspecting 1 file
C

Offenses:

test.rb:1:1: C: Style/ConditionalAssignment: Use the return of the conditional for variable assignment and comparison.
if condcondcondcondcondcondcondcondcondcondcondcondcondcondcondcondcondcondcond ...
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

1 file inspected, 1 offense detected
```


Cause
----


`Style/ConditionalAssignment` cop is aware of `Metrics/LineLength` cop's configuration.

https://github.com/bbatsov/rubocop/blob/b6d3cfed0b968f41759c20d63598f4c6764e4456/lib/rubocop/cop/style/conditional_assignment.rb#L379-L386


So, it is an expected behaviour.
